sdk/Lusid.Sdk.Tests/tutorials/Instruments/FxOptionExamples.cs

```diff
@@ -150,5 +152,110 @@ public void FxOptionPortfolioCashFlowsExample(ModelSelection.ModelEnum model, bo
+
+            // We expect exactly one cashflow associated to a cash settled option, and two cashflows for a physically settled option. Both occur at expiry.
+            Assert.That(allCashFlows.Count, Is.EqualTo(isDeliveryNotCash ? 2 : 1));
+            var cashFlowDate = allCashFlows.First().TransactionDate;
+
+            // CREATE valuation request for this portfolio consisting of the option,
+            // with valuation dates a few days before, day of and a few days after the option expiration = cashflow date.
+            var valuationRequest = TestDataUtilities.CreateValuationRequest(
+                scope,
+                portfolioCode,
+                recipeCode,
+                effectiveAt: cashFlowDate.AddDays(4),
+                effectiveFrom: cashFlowDate.AddDays(-4));
+
+            // CALL GetValuation before upserting back the cashflows. We check
+            // (1) there is no cash holdings in the portfolio prior to expiration
+            // (2) that when the option has expired, the PV is zero.
+            var valuationBeforeAndAfterExpirationOption = _aggregationApi.GetValuation(valuationRequest);
+            TestDataUtilities.CheckNoCashPositionsInValuationResults(
+                valuationBeforeAndAfterExpirationOption,
+                option.DomCcy);
+            TestDataUtilities.CheckNonZeroPvBeforeMaturityAndZeroAfter(
+                valuationBeforeAndAfterExpirationOption,
+                option.OptionSettlementDate);
+
+            // UPSERT the cashflows back into LUSID. We first populate the cashflow transactions with unique IDs.
+            var upsertCashFlowTransactions = PortfolioCashFlows.PopulateCashFlowTransactionWithUniqueIds(
+                allCashFlows);
+
+            _transactionPortfoliosApi.UpsertTransactions(
+                scope,
+                portfolioCode,
+                PortfolioCashFlows.MapToCashFlowTransactionRequest(upsertCashFlowTransactions));
+
+            // HAVING upserted cashflow into LUSID, we call GetValuation again.
+            var valuationAfterUpsertingCashFlows = _aggregationApi.GetValuation(valuationRequest);
+
+            // ASSERT that we have the expected currencies of cash in the portfolio
+            var currenciesInPortfolioAfterUpsertion = valuationAfterUpsertingCashFlows
+                .Data
+                .Select(d => (string)d[TestDataUtilities.Luid])
+                .Where(luid => luid.StartsWith("CCY_"))
+                .Distinct();
+            Assert.That(currenciesInPortfolioAfterUpsertion.Count, Is.EqualTo(isDeliveryNotCash ? 2 : 1));
+
+            // ASSERT portfolio PV is constant for each valuation date.
+            // We expect this to be true since we upserted the cashflows back in.
+            // That is instrument pv + cashflow = option payoff = constant for each valuation date.
+            TestDataUtilities.CheckPvIsConstantAcrossDatesWithinTolerance(valuationAfterUpsertingCashFlows, relativeDifferenceTolerance: 1e-10);
```

Comment:
may worth adding (again) comment that this is because fx rate is const.

Reply:
done